### ISSUE TITLE
Support for ES6 Classes

### ISSUE THREAD

**Issue by nevcos** [bluetech/ng-annotate-patched#4]
*ng-annotate-patched* should support ES6 Classes in the following way:

```
/** @ngInject */
class Test1_0 {
  constructor($log) {}
}
class Test1_1 {
  /** @ngInject */
  constructor($log) {}
}
/** @ngInject */
class Test1_2 {
  constructor($log) {
    "ngInject";
  }
}

// ========================================

/** @ngInject */
var Test2_0 = class {
  constructor($log) {}
};
var Test2_1 = /** @ngInject */ class {
  constructor($log) {}
};
var Test2_2 = class {
  /** @ngInject */
  constructor($log) {}
};
var Test2_3 = class {
  constructor($log) {
    "ngInject";
  }
};

// ========================================

var Test3_0, Test3_1, Test3_2, Test3_3;
/** @ngInject */
Test3_0 = class {
  constructor($log) {}
};
Test3_1 = /** @ngInject */ class {
  constructor($log) {}
};
Test3_2 = class {
  /** @ngInject */
  constructor($log) {}
};
Test3_3 = class {
  constructor($log) {
    "ngInject";
  }
};
```

And it should produce the correct $inject array for each scenario.

**Comment by nevcos** [comment]
@bluetech FYI I'm working on this.

I'm currently able to correctly produce annotations for:
- Declarative style with `/* @ngInject */` on the class - Tests 1_0 / 2_0 / 3_0
- Declarative style with `"ngInject"` on the constructor - Tests 1_2 / 2_3 / 3_3
- Expression style with `/* @ngInject */` on the class - Tests 2_1 / 3_1

The one that is still lacking is using the `/* @ngInject */` on constructor, will work on this tomorrow.

Please tell me if I'm missing something.

Thanks!

**Comment by bluetech** [comment]
Looking forward to it!

For what it's worth, my personal use case is only to support ES5 syntax (transpiled from ES6) + ES6 modules (not transpiled), that's why I never tried to add support for the rest of ES6 syntax. But will be great if we can add it.

**Comment by bluetech** [comment]
Oh, and I hope none of the commits I just pushed cause any conflicts for your work. If so, I apologize!

**Comment by nevcos** [comment]
@bluetech great work on removing the build process and embracing ES6 on the project 👍! Looks much better now.

About this issue, I've created PR bluetech/ng-annotate-patched/pull/5 that adds ES6 classes support.
Please check if everything is looking OK.

Thanks!
